matrix_sdk/src/client.rs

```diff
@@ -679,8 +714,205 @@ impl Client {
 
         let request = assign!(
             login::Request::new(
-                login::UserInfo::MatrixId(user),
-                login::LoginInfo::Password { password },
+                login::LoginInfo::Password { identifier: login::UserIdentifier::MatrixId(user), password },
+            ), {
+                device_id: device_id.map(|d| d.into()),
+                initial_device_display_name,
+            }
+        );
+
+        let response = self.send(request, None).await?;
+        self.base_client.receive_login_response(&response).await?;
+
+        Ok(response)
+    }
+
+    /// Login to the server via Single Sign-On.
+    ///
+    /// This takes care of the whole SSO flow:
```

Comment:
Let's mention the security implication of this, users with root access have the ability to snoop in on the data/token that is passed to the local HTTP server that will be spawned.